Add Firebolt AI demo

mohtashin opened the pull request:
This PR introduces a fully-functional RAG Chatbot Demo that shows how to pair Firebolt with an LLM (OpenAI) and a lightweight UI (Chainlit). The demo lets users ask natural-language questions and returns answers backed by live SQL queries against Firebolt. It’s designed to be copy-paste-friendly so anyone can spin it up in minutes.

Commit:
Add Firebolt AI demo

fb-cole commented on fb_ai_demo/README.md:
```diff
@@ -0,0 +1,147 @@
+# Firebolt AI Chatbot Demo
+
+A reference implementation of a **Retrieval-Augmented Generation (RAG) chatbot** powered by [Chainlit](https://docs.chainlit.io/) for the chat UI, **OpenAI** for the LLM, and **Firebolt** as the analytical database and vector store. It lets you ask questions in everyday language and receive answers backed by SQL queries that run live on Firebolt.
```
Let's make this a complete sentence. "This is a..."

fb-cole commented on fb_ai_demo/README.md:
````diff
@@ -0,0 +1,147 @@
+FIREBOLT_ENGINE_NAME="..."
+```
+
+1. **Save & close** the file—Chainlit will pick it up automatically.
````
Maybe make this more conversational, a 2-item list is weird. "Once you're done editing, save and close the file. Optionally, you can fine-tune..."

fb-cole commented on fb_ai_demo/README.md:
```diff
@@ -0,0 +1,147 @@
+
+Try queries like:
+
+* *“How many clicks were driven by each search engine in **Q4 2024**?”*
+* *“Refine a query that finds the **top 10 keywords** for Google in that period.”*
+* *“Show me a bar chart of clicks per engine.”*
```
Because this demo is made generic, you should reframe these questions and provide instructions for how to best prompt on any dataset.

Maybe also state something about ensuring your Firebolt schema has well-named columns, e.g. "col1" and "col2" will leave OpenAI clueless as to what data it's working with.

fb-cole commented on fb_ai_demo/README.md:
```diff
@@ -0,0 +1,147 @@
+
+---
+
+## 7 – Roadmap & Ideas
+
+* **Streaming** partial LLM responses
+* **Explain Plan** tab that shows Firebolt’s query plan
+* Plug-in system for domain-specific tools (e.g. Jira, Slack)
+* One-click deploy to **Render** / **Fly.io**
+* OAuth login to protect prod instances
```
I think this is a little ambitious - unless you're planning on implementing any of these later this summer, we can just strip it out.

Commit:
fix: read.me suggestion

Commit:
Update with changes on section 5

Commit:
Update AI demo README based on PR feedback

- Make opening sentence complete
- Fix clone URL to point to correct repo
- Simplify configuration instructions
- Replace template-driven prompting section with natural language guidance
- Add complete venv setup instructions
- Update gitignore to exclude Claude materials
